idx/cassandra/cassandra.go

```diff
@@ -241,29 +243,38 @@ func (c *CasIdx) Init() error {
 	c.rebuildIndex()
 
 	if memory.IndexRules.Prunable() {
+		c.wg.Add(1)
 		go c.prune()
 	}
 
 	if memory.MetaTagSupport {
+		c.wg.Add(1)
 		go c.pollStore()
 		if c.Config.updateCassIdx {
+			c.wg.Add(1)
 			go c.pruneMetaRecords()
 		}
 	}
 
+	if c.Config.ConnectionCheckInterval > 0 {
+		log.Infof("cassandra-idx: dead connection check enabled with an interval of %s", c.Config.ConnectionCheckInterval.String())
+		c.wg.Add(1)
+		go c.Session.DeadConnectionRefresh(&c.wg)
+	}
```

Comment:
Wouldn't it be better to move this whole part into `cassandra.NewSession()`? Then the user of the session doesn't need to worry about whether they started the dead connection refresh method etc. I'm also not convinced that there is a need for this wait group that gets passed into `DeadConnectionRefresh()` because since `cassandra.Session` is in a completely different package it won't modify anything inside the state of the user (cassandra store / index) anyway, so it can just shutdown independently if `c.Session.Stop()` is called. Otherwise, by sharing the wait group and the shut down channel between the cassandra store/idx and the `cassandra.Session` struct those things just get unnecessarily inter-twined, f.e. what if another future user doesn't use a shutdown chan?

Reply:
Yeah we can move all of that into `Session` and then add a `Shutdown` method.

Reply:
i pushed a commit implementing my suggestion